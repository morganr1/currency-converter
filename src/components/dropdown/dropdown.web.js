import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';

const useStyles = makeStyles(theme => ({
  formControl: {
    margin: theme.spacing(1)
  },
  select: {
    width: 170,
    [theme.breakpoints.down(460)]: {
      width: 120
    },
    [theme.breakpoints.down(362)]: {
      width: 92
    }
  },
  selectEmpty: {
    marginTop: theme.spacing(2)
  }
}));

export const DropDownComponent = ({
  disabled,
  values,
  currentValue,
  onChangeCallback,
  label
}) => {
  const classes = useStyles();
  const inputLabel = React.useRef(null);
  const [labelWidth, setLabelWidth] = React.useState(0);

  React.useEffect(() => {
    setLabelWidth(inputLabel.current.offsetWidth);
  }, []);

  return (
    <Fragment>
      <FormControl variant="outlined" className={classes.formControl}>
        <InputLabel ref={inputLabel} id={`${label}-dropdown-label`}>
          {label}
        </InputLabel>
        <Select
          disabled={disabled}
          className={classes.select}
          name={label}
          labelId={`${label}-dropdown-label`}
          id={`${label}-dropdown`}
          value={currentValue}
          onChange={onChangeCallback}
          labelWidth={labelWidth}
        >
          {values.map(curr => (
            <MenuItem key={curr} value={curr}>
              {curr}
            </MenuItem>
          ))}
        </Select>
      </FormControl>
    </Fragment>
  );
};

DropDownComponent.propTypes = {
  disabled: PropTypes.bool,
  values: PropTypes.array.isRequired,
  currentValue: PropTypes.string.isRequired,
  onChangeCallback: PropTypes.func.isRequired,
  label: PropTypes.string.isRequired
};
