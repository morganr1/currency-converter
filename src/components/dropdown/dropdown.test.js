import React from 'react';
import Select from '@material-ui/core/Select';
import { DropDownComponent } from './dropdown.web';
import { shallow } from 'enzyme';

describe('Dropdown Component', () => {
    const mockProps = {
        values: ['EUR', 'GBP'],
        currentValue: 'EUR',
        label: 'Test Dropdown Label',
        onChangeCallback: jest.fn(),
    };
    const wrapper = shallow(<DropDownComponent{...mockProps} />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('change event calls the onChangeCallback', () => {
        const selectComp = wrapper.find(Select);
        expect(selectComp).toBeDefined();
        selectComp.simulate('change', { target: { value: 'GBP' }});
        expect(mockProps.onChangeCallback).toHaveBeenCalledWith({ target: { value: 'GBP' }});
    });
});