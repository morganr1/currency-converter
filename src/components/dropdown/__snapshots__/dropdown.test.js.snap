// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component renders correctly 1`] = `
<Fragment>
  <WithStyles(ForwardRef(FormControl))
    className="makeStyles-formControl-1"
    variant="outlined"
  >
    <WithStyles(ForwardRef(InputLabel))
      id="Test Dropdown Label-dropdown-label"
    >
      Test Dropdown Label
    </WithStyles(ForwardRef(InputLabel))>
    <WithStyles(ForwardRef(Select))
      className="makeStyles-select-2"
      id="Test Dropdown Label-dropdown"
      labelId="Test Dropdown Label-dropdown-label"
      labelWidth={0}
      name="Test Dropdown Label"
      onChange={[MockFunction]}
      value="EUR"
    >
      <WithStyles(ForwardRef(MenuItem))
        key="EUR"
        value="EUR"
      >
        EUR
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        key="GBP"
        value="GBP"
      >
        GBP
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </WithStyles(ForwardRef(FormControl))>
</Fragment>
`;
